src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ProcessTests.cs

```diff
@@ -145,7 +145,137 @@ public async Task MultiProcessIdentificationTest(DiagnosticPortConnectionMode mo
         }
 
         /// <summary>
-        /// Verifies that a process was found in the identifiers list and checks the /processes/{processKey} route for the same process.
+        /// Tests that HTTP Requests using PID, UID, and/or Name have the correct response
+        /// </summary>
+        [Theory]
+        [InlineData(DiagnosticPortConnectionMode.Connect)]
+#if NET5_0_OR_GREATER
+        [InlineData(DiagnosticPortConnectionMode.Listen)]
+#endif
+        public async Task ProcessRequestConsistencyTest(DiagnosticPortConnectionMode mode)
+        {
+            DiagnosticPortHelper.Generate(
+                mode,
+                out DiagnosticPortConnectionMode appConnectionMode,
+                out string diagnosticPortPath);
+
+            await using MonitorRunner toolRunner = new(_outputHelper);
+            toolRunner.ConnectionMode = mode;
+            toolRunner.DiagnosticPortPath = diagnosticPortPath;
+            toolRunner.DisableAuthentication = true;
+            await toolRunner.StartAsync();
+
+            using HttpClient httpClient = await toolRunner.CreateHttpClientDefaultAddressAsync(_httpClientFactory);
+            ApiClient apiClient = new(_outputHelper, httpClient);
+
+            const int appCount = 1; // May add more extensive testing in the future that utilizes multiple processes
+            AppRunner[] appRunners = new AppRunner[appCount];
+
+            for (int i = 0; i < appCount; i++)
+            {
+                AppRunner runner = new(_outputHelper, Assembly.GetExecutingAssembly(), appId: i + 1);
+                runner.ConnectionMode = appConnectionMode;
+                runner.DiagnosticPortPath = diagnosticPortPath;
+                runner.ScenarioName = TestAppScenarios.AsyncWait.Name;
+                runner.Environment[ExpectedEnvVarName] = Guid.NewGuid().ToString("D");
+
+                appRunners[i] = runner;
+            }
+
+            IEnumerable<ProcessIdentifier> identifiers;
```

Comment:
This can go inside the ExceuteAsync annonymous function since it is not used outside.